### PR TITLE
update to reflect new syntect_server version; improve logging

### PR DESCRIPTION
**Pull request by slimsag**
- Update to reflect new HA syntect_server version.
- Log highlighting failures regardless of cause for better debugging.

Test plan: Manual testing, currently.

Helps https://github.com/sourcegraph/sourcegraph/issues/5406

### CODE REVIEW

File: pkg/highlight/highlight.go [hunk 1]
```diff
@@ -158,19 +169,23 @@ func Code(ctx context.Context, p Params) (h template.HTML, aborted bool, err err
...
-				"repo_name", p.Metadata.RepoName,
-				"revision", p.Metadata.Revision,
-				"snippet", fmt.Sprintf("%q…", firstCharacters(code, 80)),
-				"error", err,
-			)
 			if cause == gosyntect.ErrPanic {
 				tr.LogFields(otlog.Bool("panic", true))
 				prometheusStatus = "panic"
 			}
+			if cause == gosyntect.ErrHSSWorkerTimeout {
```
Comment (tsenart) [inline pkg/highlight/highlight.go]: Is this ever true inside this if block? The condition in line [180](https://github.com/sourcegraph/sourcegraph/pull/5842/files#diff-5a881b36b33270d5031b9a3060269516R180) tests for two specific error causes, and this one is not in that set.

Comment (tsenart) [reply pkg/highlight/highlight.go]: Suggestion: Perhaps this would be easier to read with a switch statement.

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,6 +27,12 @@ All notable changes to Sourcegraph are documented in this file.
 
 ### Removed
 
+## 3.8.2
+
+### Fixed
+
+- Sourcegraph cluster deployments now run a more stable syntax highlighting server which can self-recover from rarer failure cases such as getting stuck at high CPU usage when highlighting some specific files. This will be ported to single-container deployments [at a later date](https://github.com/sourcegraph/sourcegraph/issues/5841).
```
Comment (keegancsmith) [inline CHANGELOG.md]: what is the reason to do a patch release vs waiting for 3.9?

File: pkg/highlight/highlight.go (hunk 1 above)
Comment (slimsag) [reply pkg/highlight/highlight.go]: Good catch and yes, definitely. Thank you! Fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (slimsag) [reply CHANGELOG.md]: I updated this entry to include a link to the issue, but the explanation here is that this is actually a quite serious issue and is causing some instability for https://app.hubspot.com/contacts/2762526/company/407948923 -- see https://github.com/sourcegraph/sourcegraph/issues/5406 for details

**Pull request merged**